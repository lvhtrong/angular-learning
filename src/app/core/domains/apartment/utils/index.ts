import { Apartment as StoreApartment } from '@store/state/domains/apartment/apartment.model';
import { ApartmentDTO } from '../models/apartment.dto';

export function mapToStoreApartment(apartment: ApartmentDTO): StoreApartment {
  return {
    id: apartment.id,
    updatedAt: apartment.updatedAt,
    description: apartment.description,
    price: apartment.price,
    address: apartment.address,
    location: apartment.location,
    specs: apartment.specs,
    images: apartment.images,
    utilities: apartment.utilities,
    banner: apartment.banner,
    title: apartment.title,
    type: apartment.type,
    createdBy: apartment.createdBy,
    createdAt: apartment.createdAt,
    roomsCount: apartment.roomsCount,
    rooms: apartment.rooms,
  };
}

export function mapFromStoreApartment(apartment: StoreApartment): ApartmentDTO {
  return {
    id: apartment.id,
    updatedAt: apartment.updatedAt,
    description: apartment.description,
    price: apartment.price,
    address: apartment.address,
    location: apartment.location,
    specs: apartment.specs,
    images: apartment.images,
    utilities: apartment.utilities,
    banner: apartment.banner,
    title: apartment.title,
    type: apartment.type,
    createdBy: apartment.createdBy,
    createdAt: apartment.createdAt,
    roomsCount: apartment.roomsCount,
    rooms: apartment.rooms,
  };
}
