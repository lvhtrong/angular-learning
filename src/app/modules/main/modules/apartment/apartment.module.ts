import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';

import { ApartmentRoutingModule } from './apartment-routing.module';
import { ListComponent } from './pages/list/list.component';

@NgModule({
  declarations: [ListComponent],
  imports: [CommonModule, ApartmentRoutingModule],
})
export class ApartmentModule {}
