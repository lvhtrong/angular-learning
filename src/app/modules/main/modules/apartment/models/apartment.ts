export interface Apartment {
  id: string;
  title?: string;
  type?: string;
  shortDescription?: string;
  address?: string;
}
