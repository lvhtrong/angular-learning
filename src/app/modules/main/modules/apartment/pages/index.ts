import { ListComponent } from './list/list.component';

export const pages = [ListComponent];
