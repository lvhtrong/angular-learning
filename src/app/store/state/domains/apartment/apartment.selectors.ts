import { createFeatureSelector, createSelector } from '@ngrx/store';
import { selectDomains } from '../domains.selectors';

export const selectApartment = createSelector(
  selectDomains,
  (state) => state.apartments
);

export const selectApartmentEntities = createSelector(
  selectApartment,
  (state) => state.entities
);
