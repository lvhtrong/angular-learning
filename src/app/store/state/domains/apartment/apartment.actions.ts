import { createAction, props } from '@ngrx/store';
import { Update } from '@ngrx/entity';

import { Apartment } from './apartment.model';

export const loadApartments = createAction(
  '[Apartment/API] Load Apartments',
  props<{ apartments: Apartment[] }>()
);

export const addApartment = createAction(
  '[Apartment/API] Add Apartment',
  props<{ apartment: Apartment }>()
);

export const upsertApartment = createAction(
  '[Apartment/API] Upsert Apartment',
  props<{ apartment: Apartment }>()
);

export const addApartments = createAction(
  '[Apartment/API] Add Apartments',
  props<{ apartments: Apartment[] }>()
);

export const upsertApartments = createAction(
  '[Apartment/API] Upsert Apartments',
  props<{ apartments: Apartment[] }>()
);

export const updateApartment = createAction(
  '[Apartment/API] Update Apartment',
  props<{ apartment: Update<Apartment> }>()
);

export const updateApartments = createAction(
  '[Apartment/API] Update Apartments',
  props<{ apartments: Update<Apartment>[] }>()
);

export const deleteApartment = createAction(
  '[Apartment/API] Delete Apartment',
  props<{ id: string }>()
);

export const deleteApartments = createAction(
  '[Apartment/API] Delete Apartments',
  props<{ ids: string[] }>()
);

export const clearApartments = createAction('[Apartment/API] Clear Apartments');
