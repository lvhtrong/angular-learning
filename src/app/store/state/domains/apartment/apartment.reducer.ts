import { Action, createReducer, on } from '@ngrx/store';
import { EntityState, EntityAdapter, createEntityAdapter } from '@ngrx/entity';
import { UpdateTracking } from '@store/models/common/update-tracking.model';
import { Apartment } from './apartment.model';
import * as ApartmentActions from './apartment.actions';

export const apartmentsFeatureKey = 'apartments';

export interface State extends EntityState<Apartment>, UpdateTracking {
  // additional entities state properties
}

export const adapter: EntityAdapter<Apartment> = createEntityAdapter<Apartment>();

export const initialState: State = adapter.getInitialState({
  // additional entity state properties
  updatedAt: null
});


export const reducer = createReducer(
  initialState,
  on(ApartmentActions.addApartment,
    (state, action) => adapter.addOne(action.apartment, state)
  ),
  on(ApartmentActions.upsertApartment,
    (state, action) => adapter.upsertOne(action.apartment, state)
  ),
  on(ApartmentActions.addApartments,
    (state, action) => adapter.addMany(action.apartments, state)
  ),
  on(ApartmentActions.upsertApartments,
    (state, action) => adapter.upsertMany(action.apartments, state)
  ),
  on(ApartmentActions.updateApartment,
    (state, action) => adapter.updateOne(action.apartment, state)
  ),
  on(ApartmentActions.updateApartments,
    (state, action) => adapter.updateMany(action.apartments, state)
  ),
  on(ApartmentActions.deleteApartment,
    (state, action) => adapter.removeOne(action.id, state)
  ),
  on(ApartmentActions.deleteApartments,
    (state, action) => adapter.removeMany(action.ids, state)
  ),
  on(ApartmentActions.loadApartments,
    (state, action) => adapter.setAll(action.apartments, state)
  ),
  on(ApartmentActions.clearApartments,
    state => adapter.removeAll(state)
  ),
);


export const {
  selectIds,
  selectEntities,
  selectAll,
  selectTotal,
} = adapter.getSelectors();
