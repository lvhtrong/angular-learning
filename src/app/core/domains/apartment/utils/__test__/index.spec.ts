import * as utils from '../index';
import { ApartmentDTO } from '../../models/apartment.dto';
import { Apartment } from '@store/state/domains/apartment/apartment.model';

describe('mapToStoreApartment', () => {
  const getUtil = () => {
    return utils.mapToStoreApartment;
  };

  it('should return new object with structure', () => {
    const input = {
      description: {
        long: 'long-desc',
        short: 'short-desc',
      },
      price: {
        min: 1,
        max: 2,
      },
      address: {
        city: 'city-1',
        district: 'district-1',
        street: 'street-1',
      },
      location: {
        lat: 'lat-1',
        long: 'long-1',
      },
      specs: ['specs-1'],
      images: ['utilities-1'],
      utilities: ['utilities-1'],
      banner: 'banner',
      title: 'title',
      type: 'type',
      createdBy: 'createdBy',
      createdAt: 'createdAt',
      updatedAt: 'updatedAt',
      roomsCount: 2,
      id: 'id',
      rooms: ['room1', 'room2'],
    };

    const util = getUtil();
    const result = util(input);

    expect(result).toMatchSnapshot();
  });
});

describe('mapFromStoreApartment', () => {
  const getUtil = () => {
    return utils.mapFromStoreApartment;
  };

  it('should return new object with structure', () => {
    const input = {
      description: {
        long: 'long-desc',
        short: 'short-desc',
      },
      price: {
        min: 1,
        max: 2,
      },
      address: {
        city: 'city-1',
        district: 'district-1',
        street: 'street-1',
      },
      location: {
        lat: 'lat-1',
        long: 'long-1',
      },
      specs: ['specs-1'],
      images: ['utilities-1'],
      utilities: ['utilities-1'],
      banner: 'banner',
      title: 'title',
      type: 'type',
      createdBy: 'createdBy',
      createdAt: 'createdAt',
      updatedAt: 'updatedAt',
      roomsCount: 2,
      id: 'id',
      rooms: ['room1', 'room2'],
    };

    const util = getUtil();
    const result = util(input);

    expect(result).toMatchSnapshot();
  });
});
