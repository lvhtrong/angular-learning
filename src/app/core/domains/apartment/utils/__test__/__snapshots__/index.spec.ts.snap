// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mapFromStoreApartment should return new object with structure 1`] = `
Object {
  "address": Object {
    "city": "city-1",
    "district": "district-1",
    "street": "street-1",
  },
  "banner": "banner",
  "createdAt": "createdAt",
  "createdBy": "createdBy",
  "description": Object {
    "long": "long-desc",
    "short": "short-desc",
  },
  "id": "id",
  "images": Array [
    "utilities-1",
  ],
  "location": Object {
    "lat": "lat-1",
    "long": "long-1",
  },
  "price": Object {
    "max": 2,
    "min": 1,
  },
  "rooms": Array [
    "room1",
    "room2",
  ],
  "roomsCount": 2,
  "specs": Array [
    "specs-1",
  ],
  "title": "title",
  "type": "type",
  "updatedAt": "updatedAt",
  "utilities": Array [
    "utilities-1",
  ],
}
`;

exports[`mapToStoreApartment should return new object with structure 1`] = `
Object {
  "address": Object {
    "city": "city-1",
    "district": "district-1",
    "street": "street-1",
  },
  "banner": "banner",
  "createdAt": "createdAt",
  "createdBy": "createdBy",
  "description": Object {
    "long": "long-desc",
    "short": "short-desc",
  },
  "id": "id",
  "images": Array [
    "utilities-1",
  ],
  "location": Object {
    "lat": "lat-1",
    "long": "long-1",
  },
  "price": Object {
    "max": 2,
    "min": 1,
  },
  "rooms": Array [
    "room1",
    "room2",
  ],
  "roomsCount": 2,
  "specs": Array [
    "specs-1",
  ],
  "title": "title",
  "type": "type",
  "updatedAt": "updatedAt",
  "utilities": Array [
    "utilities-1",
  ],
}
`;
